internal/langserver/handlers/did_change_watched_files.go

```diff
@@ -0,0 +1,71 @@
+package handlers
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/creachadair/jrpc2"
+	"github.com/hashicorp/terraform-ls/internal/job"
+	ilsp "github.com/hashicorp/terraform-ls/internal/lsp"
+	"github.com/hashicorp/terraform-ls/internal/protocol"
+	lsp "github.com/hashicorp/terraform-ls/internal/protocol"
+	"github.com/hashicorp/terraform-ls/internal/terraform/ast"
+)
+
+func (svc *service) DidChangeWatchedFiles(ctx context.Context, params lsp.DidChangeWatchedFilesParams) error {
+	var ids job.IDs
+
+	for _, change := range params.Changes {
+		uri := string(change.URI)
+		_, err := os.Stat(uri)
+		if err != nil {
+			jrpc2.ServerFromContext(ctx).Notify(ctx, "window/showMessage", &lsp.ShowMessageParams{
+				Type:    lsp.Warning,
+				Message: fmt.Sprintf("Unable to update: %s, Failed to open directory", uri),
+			})
+			continue
+		}
+
+		// `uri` can either be a file or a director baed on the spec.
+		// We're not making any assumptions on the above and passing
+		// the uri as the module path itself for validation.
+		//
+		// https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_didChangeWatchedFiles
+		if !ast.IsModuleFilename(uri) && !ast.IsVarsFilename(uri) {
+			jrpc2.ServerFromContext(ctx).Notify(ctx, "window/showMessage", &lsp.ShowMessageParams{
+				Type:    lsp.Warning,
+				Message: fmt.Sprintf("Unable to update file: %s, filetype not supported", uri),
+			})
```

Comment:
If this isn't a Terraform file, why is it an error that we should send to the user? Isn't this a case where we ignore the file if it's not a file we can do something with? As with the file access error above, I'd rather log here than interrupt the user with an error message they have no control over.

Reply:
File access I'd be more inclined to ignore but the thing here is that clients shouldn't be sending unsupported files, so this clearly indicates client-side bug, which we'd like the user to report. We can make that clearer in the message though.

Reply:
@radeksimko do you think this can be solved if and when we have the clients have the ability to specify file patterns to watch for? Not maybe for this change right away but longer term. I'm not sure but I feel aligned with what @jpogran mentioned above. Having to show errors to the user and expect them to report them to the client seems counterintuitive. What do you think?